### DISCUSSION TITLE
Installing PlatformIO IDE from marketplace

### DISCUSSION THREAD

**Discussion by flexmatics** [coder/code-server#6016] [Q&A] [answered]
Hi,

It seems that I'm missing something obvious, but can anyone guide me what should I do to get Platformio IDE installed into code-server? Currently, it is not being found in the marketplace. Thanks.

Using this version from LXC container:
code-server: v4.9.1
Code: 1.73.1
Commit: 6261075646f055b99068d3688932416f2346dd3b

**Comment by code-asher** [comment, marked as answer]
It looks like there is a licensing issue preventing it from being added to the marketplace: https://github.com/platformio/platformio-vscode-ide/issues/1802#issuecomment-930154036

**Comment by lg-nineoneninethreenine** [comment]
I think I have an idea to get Platformio onto code-server. This is a trick that has to work. So, @flexmatics you first have to go to the VS Code Marketplace in your browser and head to Platformio's extension page. Once you are there, hit F3 and search "download". Find the "Download VSIX" link, then get the VSIX file onto your computer. Push it to your container(use any method you want, you can try `scp` too) and run this command:
```shell
code-server --install-extension <name-of-vsix-file>.vsix
```
Now I bet Platformio will have to be installed. If not, then it could mean Platformio has compatibility issues with code-server.

**Comment by flexmatics** [comment]
Thanks both for suggestions.
